planner/core/point_get_plan.go

```diff
@@ -172,17 +180,37 @@ func (p *PointGetPlan) SetOutputNames(names types.NameSlice) {
 	p.outputNames = names
 }
 
+// GetCost returns cost of the PointGetPlan.
+func (p *PointGetPlan) GetCost(cols []*expression.Column) float64 {
+	sessVars := p.ctx.GetSessionVars()
+	var rowSize float64
+	p.cost = 0
+	if p.IndexInfo == nil {
+		rowSize = p.stats.HistColl.GetTableAvgRowSize(p.ctx, cols, kv.TiKV, true)
+	} else {
+		rowSize = p.stats.HistColl.GetIndexAvgRowSize(p.ctx, cols, p.IndexInfo.Unique)
+	}
+	p.cost += rowSize * sessVars.NetworkFactor
+	p.cost += sessVars.SeekFactor
+	p.cost /= float64(sessVars.DistSQLScanConcurrency)
+	return p.cost
+}
+
 // BatchPointGetPlan represents a physical plan which contains a bunch of
 // keys reference the same table and use the same `unique key`
 type BatchPointGetPlan struct {
 	baseSchemaProducer
 
+	ctx              sessionctx.Context
 	TblInfo          *model.TableInfo
 	IndexInfo        *model.IndexInfo
 	Handles          []int64
 	HandleParams     []*driver.ParamMarkerExpr
 	IndexValues      [][]types.Datum
 	IndexValueParams [][]*driver.ParamMarkerExpr
+	KeepOrder        bool
+	Desc             bool
+	cost             float64
```

Comment:
Why is `cost` added here?

Reply:
Because we need it to avoid recalculate the cost.